src/telemetry/impression_metrics.yaml

```diff
@@ -0,0 +1,114 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+---
+$schema: moz://mozilla.org/schemas/glean/metrics/2-0-0
+
+# "impression" category events, are events where the user has just moved on to page X
+# Events under this category are recorded when a new screen is loaded
+impression:
+  main_screen:
+    type: event
+    lifetime: ping
+    send_in_pings:
+      - main
+    description: |
+      The user has just moved on to the main screen
+      i.e. the screen with the VPN toggle button.
+    bugs:
+      - https://mozilla-hub.atlassian.net/browse/VPN-5718
+    data_reviews:
+      - TODO
+    data_sensitivity:
+      - interaction
+    notification_emails:
+      - brizental@mozilla.com
+      - vpn-telemetry@mozilla.com
+    expires: never
+    extra_keys: &impression_extra_keys
+      screen:
+        description: The id of the screen the user is on
+        type: string
+  speed_test_loading_screen:
+    type: event
+    lifetime: ping
+    send_in_pings:
+      - main
+    description: |
+      The user has just moved on to the screen that shows
+      the loading animation for the speed test results
+    bugs:
+      - https://mozilla-hub.atlassian.net/browse/VPN-5366
+      - https://mozilla-hub.atlassian.net/browse/VPN-5718
+    data_reviews:
+      - https://github.com/mozilla-mobile/mozilla-vpn-client/pull/8060#issuecomment-1724272061
+      - TODO
+    data_sensitivity:
+      - interaction
+    notification_emails:
+      - brizental@mozilla.com
+      - vpn-telemetry@mozilla.com
+    expires: never
+    extra_keys: *impression_extra_keys
+  speed_test_completed_screen:
+    type: event
+    lifetime: ping
+    send_in_pings:
+      - main
+    description: |
+      The user has just moved on to the screen that shows
+      the speed test results
+    bugs:
+      - https://mozilla-hub.atlassian.net/browse/VPN-5366
+      - https://mozilla-hub.atlassian.net/browse/VPN-5718
+    data_reviews:
+      - https://github.com/mozilla-mobile/mozilla-vpn-client/pull/8060#issuecomment-1724272061
+      - TODO
+    data_sensitivity:
+      - interaction
+    notification_emails:
+      - brizental@mozilla.com
+      - vpn-telemetry@mozilla.com
+    expires: never
+    extra_keys: *impression_extra_keys
+  speed_test_error_screen:
+    type: event
+    lifetime: ping
+    send_in_pings:
+      - main
+    description: |
+      The user has just moved on to the screen that shows
+      the speed test results
```

Comment:
This description seems to be incorrect for the speed_test_error_screen. 

Also, just a nit, the user was moved to this error screen by the app, so should all these descriptions say something like "Moved to screen that shows ...", which handles the case where the screen was navigated to by the app or explicitly by the user?  Feel free to ignore this suggestion.

Reply:
I tried the change of wording, but I thought it looked weird 😅 I'll leave it as is. This is simple enough to change in the future if it becomes confusing.